src/renderer/features/dynamicDerivations/ImportKeysModal/ImportKeysModal.tsx

```diff
@@ -0,0 +1,111 @@
+import { useUnit } from 'effector-react';
+import { useEffect } from 'react';
+
+import { Alert, BaseModal, Button, InfoLink, InputFile, InputHint } from '@renderer/shared/ui';
+import { useI18n } from '@renderer/app/providers';
+import { importKeysModel, TypedImportedDerivation } from '@renderer/entities/dynamicDerivations';
+import { AccountId } from '@renderer/shared/core';
+import { cnTw } from '@renderer/shared/lib/utils';
+// @ts-ignore
+import templateFile from '@renderer/assets/files/dd-template.yaml';
+
+type Props = {
+  isOpen: boolean;
+  rootAccountId: AccountId;
+  existingKeys: TypedImportedDerivation[];
+  onClose: () => void;
+};
+
+export const ImportKeysModal = ({ isOpen, onClose, rootAccountId, existingKeys }: Props) => {
+  const { t } = useI18n();
+  const validationError = useUnit(importKeysModel.$validationError);
+  const successReport = useUnit(importKeysModel.$successReport);
+
+  useEffect(() => {
+    if (rootAccountId && existingKeys) {
+      importKeysModel.events.importStarted({ derivations: existingKeys, root: rootAccountId });
+    }
+  }, [rootAccountId, existingKeys]);
+
+  const handleFileUpload = (file: File) => {
+    const fileReader = new FileReader();
+    fileReader.onloadend = (e) => {
+      if (e.target?.result) {
+        importKeysModel.events.fileUploaded(e.target['result'] as string);
+      }
+    };
+
+    fileReader.readAsText(file);
```

Comment:
Do I really need to init `FileReader` and wait for `onloadend` event?
Or some additional actions required when reading `text (yaml)` file?
Never seen that before, maybe you guys can help @Asmadek @sokolova-an ?